core/src/test/java/projectzero/UMLClassYamlMapperTest.java

```diff
@@ -0,0 +1,33 @@
+package projectzero;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class UMLClassYamlMapperTest {
+
+    @Test
+    void testReadsAndWritesClassesToFile(@TempDir Path tempDir) throws IOException {
+        Path path = tempDir.resolve("animals.yaml");
+
+        UMLClass dog = new UMLClass("dog");
+        UMLClass cat = new UMLClass("cat");
+        UMLClass bat = new UMLClass("bat");
+
+        List<UMLClass> animals = Arrays.asList(dog, cat, bat);
+
+        UMLClassYamlMapper umlClassYamlMapper = new UMLClassYamlMapper();
+        umlClassYamlMapper.write(path.toString(), animals);
+
+        List<String> animalNames = animals.stream().map(animal -> animal.getName()).collect(Collectors.toList());
+        List<String> readAnimalNames = umlClassYamlMapper.read(path.toString()).stream().map(animal -> animal.getName()).collect(Collectors.toList());
+
+        Assertions.assertArrayEquals(animalNames.toArray(), readAnimalNames.toArray());
```

Comment:
Is this the best way to do this? Seems weird to write it like this. Mostly line 29.

Reply:
The `stream().map(animal -> animal.getName().collect(Collectors.toList())` is the Java equivalent to the python `list(map(lambda animal: animal.getName()))`. It's a little more verbose than it would be in python or JS, but it does the same thing. It's the simplest way to write it in terms of lines of code. 

I'm only doing it this way because if you do `Assertions.assertArrayEquals()` on two lists of objects, it will fail because they aren't the same object. I have to look into a way of asserting that objects equal by fields.